packages/wrangler/src/__tests__/deploy.test.ts

```diff
@@ -1689,6 +1683,81 @@ Update them to point to this script instead?`,
 			`);
 		});
 
+		it("should warn on mounted paths", async () => {
+			writeWranglerConfig({
+				routes: [
+					"simple.co.uk/path/*",
+					"simple.co.uk/*",
+					{ pattern: "example.com/blog/*", zone_id: "asdfadsf" },
+					{ pattern: "example.com/*", zone_id: "asdfadsf" },
+					{ pattern: "example.com/abc/def/*", zone_id: "asdfadsf" },
+				],
+			});
+			await mockAUSRequest([]);
+			mockSubDomainRequest();
+			mockUpdateWorkerSubdomain({ enabled: false, previews_enabled: true });
+			mockUploadWorkerRequest({
+				expectedAssets: {
+					jwt: "<<aus-completion-token>>",
+					config: {},
+				},
+				expectedType: "none",
+			});
+			mockPublishRoutesRequest({
+				routes: [
+					// @ts-expect-error - this is what is expected
+					{
+						pattern: "simple.co.uk/path/*",
+					},
+					// @ts-expect-error - this is what is expected
+					{
+						pattern: "simple.co.uk/*",
+					},
+					{
+						pattern: "example.com/blog/*",
+						zone_id: "asdfadsf",
+					},
+					{
+						pattern: "example.com/*",
+						zone_id: "asdfadsf",
+					},
+					{
+						pattern: "example.com/abc/def/*",
+						zone_id: "asdfadsf",
+					},
+				],
+			});
+
+			writeWorkerSource();
+			writeAssets([{ filePath: "asset.txt", content: "Content of file-1" }]);
+
+			await runWrangler(`deploy --assets assets`);
+
+			expect(std.warn).toMatchInlineSnapshot(`
+				"[33m▲ [43;33m[[43;30mWARNING[43;33m][0m [1mWarning: The following routes will attempt to serve Assets on a mounted path:[0m
+
+				    • simple.co.uk/path/* (Will match assets: assets/path/*)
+				    • example.com/blog/* (Will match assets: assets/blog/*)
+				    • example.com/abc/def/* (Will match assets: assets/abc/def/*)
+
+				  Requests not matching an asset will be forwarded to the Worker.
```

Comment:
```suggestion
				  Requests not matching an asset will be forwarded to the Worker script.
```

Reply:
I'd push back on script—we've been trying to move away from this language and towards Worker. To me, this is naturally a Worker + Assets situation, which makes the Worker language make sense for your code. Script is limiting, and makes it sound like workers are small scripts rather than complex fully featured applications with multiple modules & bindings

Reply:
https://github.com/cloudflare/workers-sdk/pull/7476/files#r1873572047

Yeah, I'm not super married to it. But lay out my thinking ^